### ISSUE TITLE
Default subcommand with options

### ISSUE THREAD

**Issue by oscarotero** [tj/commander.js#461]
Hi, I want to use a subcommand as default, and this subcommands has options. Something like this:

``` js
//app.js

program
    .command('run', 'Execute stuff', {isDefault: true})
    .command('list', 'List stuff')
    .parse(process.argv);
```

``` js
// app-run.js

program
    .option('-c, --config <file>', 'use a config file')
    .parse(process.argv);
```

This command works fine:

```
app run -c config.json
```

And this as well (executes the default subcommand):

```
app
```

but on execute the default subcommand with options: 

```
app -c config.json
```

I have the followin error:

```
  error: unknown option `-c'
```

**Comment by natew** [comment]
Big +1 on this

My workaround is not to use isDefault and just coerce argv, for now:

```js
// make `flint run` default command
var flintIndex = getFlintIndex()

 // find where index of flint is
function getFlintIndex() {
  let index = 0
  for (let arg of process.argv) {
    if (arg.indexOf('flint') > 0)
      return index

    index++
  }
}

// make sure flags are still passed to `flint run`
var firstFlag = process.argv[flintIndex + 1]

if (flintIndex === process.argv.length - 1 || (firstFlag && firstFlag[0] === '-')) {
  process.flintArgs = [].concat(process.argv);
  process.flintArgs.splice(flintIndex + 1, 0, 'run');
}
```

**Comment by SomeKittens** [comment]
Not sure how we'd accomplish this - you want us to skip all params if there's a default subcommand?

**Comment by natew** [comment]
No, it's just to pass those options down to the subcommand.

So `flint run --cached` works, and if `run` is default, then `flint --cached` would work the same.

**Comment by SomeKittens** [comment]
My issue is what happens if the root program (`app` in this case) also has a `-c` option.

**Comment by natew** [comment]
Ah, so could we have an option `isDefault: true, passOptions: true`

**Comment by alexcanessa** [comment]
Any updates on this one?

**Comment by shadowspawn** [comment]
Took me a while to get on board with this one, but I now agree this is the correct behaviour and interested in making it happen.

The PR in flight is #614, and an easy code change. :-)

**Comment by shadowspawn** [comment]
There are improvements on the develop branch for the next release (#614 and #1047).

**Comment by shadowspawn** [comment]
v4.0.0-0 prerelease published: #1067

**Comment by shadowspawn** [comment]
v4.0.0 has been released which hopefully resolves this issue
